src/storage/mutate/DeleteVerticesProcessor.cpp

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/mutate/DeleteVerticesProcessor.h"
+#include "base/NebulaKeyUtils.h"
+
+DECLARE_bool(enable_vertex_cache);
+
+namespace nebula {
+namespace storage {
+
+void DeleteVerticesProcessor::process(const cpp2::DeleteVerticesRequest& req) {
+    auto spaceId = req.get_space_id();
+    const auto& partVertices = req.get_parts();
+    auto iRet = indexMan_->getTagIndexes(spaceId);
+    if (iRet.ok()) {
+        for (auto& index : iRet.value()) {
+            indexes_.emplace_back(index);
+        }
+    }
+
+    if (indexes_.empty()) {
+        std::for_each(partVertices.begin(), partVertices.end(), [&](auto& pv) {
```

Comment:
why  loop twice?

Reply:
Seems couldn't merge into one loop. In each `handleAsync` will check is it finished and onFinished().